### ISSUE TITLE
'kill' kills built-in properties of operator after user-defined property

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#1834]
Imported from SourceForge on 2024-07-05 09:20:02
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2014-03-04 23:27:41
Original: https://sourceforge.net/p/maxima/bugs/2695

---

After setting a user-defined property for an operator, 'kill' kills some built-in properties instead of the user-defined property. E.g.:

    put ("*", 'foo, 'bar);
    :lisp (symbol-plist 'mtimes)
      => (bunch of stuff including display properties)
    kill (all);
    :lisp (symbol-plist 'mtimes)
      => (display properties have gone missing)
    2*x;
      => mtimes(2, x)

But the user-defined property is still there (it's supposed to be false after kill).

    get ("*", 'bar);
      => foo

This bug is tickled by 'load(solver)' followed by 'kill' but the real culprit is the interaction between 'put' and 'kill'.

Tracing some of the functions involved, I see that 'put("\*", ...)' attaches the property to the symbol '\* instead of 'MTIMES or '$* as one would expect. Maybe this is part or all of the problem.

build_info says:

    Maxima version: "5.32.1"
    Maxima build date: "2014-01-02 09:53:22"
    Host type: "i686-pc-linux-gnu"
    Lisp implementation type: "CLISP"
    Lisp implementation version: "2.49 (2010-07-07) (built 3539704948) (memory 3597674023)"

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 09:20:04
Created by **[rswarbrick](https://sourceforge.net/u/rswarbrick/)** on 2014-03-20 21:41:59
Original: https://sourceforge.net/p/maxima/bugs/2695/#b1f4

---

I haven't worked out what's going on yet, but it looks like the search can be narrowed down a little:

~~~~
:::text
put("*", 'foo, 'bar);

:lisp (symbol-plist 'mtimes)

:lisp (kill1 "*")

:lisp (symbol-plist 'mtimes)
~~~~

shows the stuff being killed. kill1 starts with a call to getopr0, which I think is what's confusing things (it doesn't explain why the display properties get nuked though...)

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 09:20:08
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2014-03-27 21:01:38
Original: https://sourceforge.net/p/maxima/bugs/2695/#b38e

---

- **labels**:  --> kill, properties
- **status**: open --> closed

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 09:20:12
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2014-03-27 21:01:38
Original: https://sourceforge.net/p/maxima/bugs/2695/#9137

---

Fixed by commit 0067e7c33 and commit 9507951ba. Closing this report.
